server/src-lib/Hasura/Server/Init/Config.hs

```diff
@@ -260,7 +261,7 @@ instance FromEnv LQ.BatchSize where
   fromEnv = fmap LQ.BatchSize . readEither
 
 instance FromEnv LQ.RefetchInterval where
-  fromEnv = fmap (LQ.RefetchInterval . milliseconds . fromInteger) . readEither
+  fromEnv = fmap (LQ.RefetchInterval . NonNegativeDiffTime . milliseconds . fromInteger) . readEither
```

Comment:
Instead of using the `NonNegativeDiffTime` constructor here, use the `mkNonNegativeDiffTime`. Currently, negative values will also be accepted.